sematic/api_client.py

```diff
@@ -495,6 +495,14 @@ def _put(
     endpoint: str,
     json_payload: Optional[Dict[str, Any]] = None,
     data: Optional[bytes] = None,
+) -> Any:
+    return _put_no_retries(endpoint=endpoint, json_payload=json_payload, data=data)
+
+
+def _put_no_retries(
+    endpoint: str,
+    json_payload: Optional[Dict[str, Any]] = None,
+    data: Optional[bytes] = None,
 ) -> Any:
     response = request(requests.put, endpoint, dict(json=json_payload, data=data))
```

Comment:
I recommend centralizing this logic in the `request` function, adding a `retry: bool` argument, and removing the `@retry` decorator from `_get`, `_post`, and `_put`:

```python
def _request(
    ..., retry: bool = True
) -> Response:

    @retry(..., tries=4 if retry else 1)
    def _do_request():
        ...

    _do_request()
    ...

```

Then propagating up the `retry` argument to `_get`, `_post`, and `_put`, with default set to `True`.

Then
```python
def cancel_resolution(resolution_id: str) -> Resolution:
    response = _put(f"/resolutions/{resolution_id}/cancel", {}, retry=False)
```

More work, but seems cleaner?